ambari-server/src/main/java/org/apache/ambari/server/serveraction/kerberos/stageutils/KerberosKeytabController.java

```diff
@@ -54,6 +58,13 @@
   @Inject
   private KerberosKeytabPrincipalDAO kerberosKeytabPrincipalDAO;
 
+  //TODO: due to circular dependencies in Guice this field cannot be injected with Guice's @Inject annotation; for now we should statically inject in AmbariServer
```

Comment:
This seems pretty bad..

Reply:
How about injecting a Provider of KerberosHelper? That would allow lazy injection

Reply:
That would also work (as described [here](https://github.com/google/guice/wiki/CyclicDependencies#break-the-cycle-with-a-provider)); but - IMO - Guice providers are just hiding cyclic dependencies which is the real issue here.
In this case it's not better or worse than the solution I applied (we have plenty of these in the code anyway); one advantage would be is to not add another static injection.

Long story short: I'll submit a new patch in 5-10 mins.